pkg/sql/crdb_internal.go

```diff
@@ -1409,6 +1432,134 @@ func makeJobsTableRows(
+LEFT OUTER JOIN system.public.job_status AS s ON j.id = s.job_id  
+	` + whereClause
+
+	isAdmin, err := p.HasAdminRole(ctx)
+	if err != nil {
+		return emitted, err
+	}
+
+	it, err := p.InternalSQLTxn().QueryIteratorEx(
+		ctx, "crdb-internal-jobs-tables", p.txn, sessiondata.InternalExecutorOverride{User: p.User()}, query, params...)
+	if err != nil {
+		return emitted, err
+	}
+	defer func() {
+		if err := it.Close(); err != nil {
+			retErr = errors.CombineErrors(retErr, err)
+		}
+	}()
+
+	sessionJobs := make([]*jobs.Record, 0, p.extendedEvalCtx.jobs.numToCreate())
+	uniqueJobs := make(map[*jobs.Record]struct{})
+	if err := p.extendedEvalCtx.jobs.forEachToCreate(func(job *jobs.Record) error {
+		if _, ok := uniqueJobs[job]; ok {
+			return nil
+		}
+		sessionJobs = append(sessionJobs, job)
+		uniqueJobs[job] = struct{}{}
+		return nil
+	}); err != nil {
+		return emitted, err
+	}
+
+	// Loop while we need to skip a row.
+	for {
+		ok, err := it.Next(ctx)
+		if err != nil {
+			return emitted, err
+		}
+		// We will read the columns from the query on joined jobs tables into a wide
+		// row, and then copy the values from read rows into named variables to then
+		// use when emitting our output row. If we need to synthesize rows for jobs
+		// pending creation in the session, we'll do so in those same named vars to
+		// keep things organized.
+		//   0,      1,    2,        3,     4,      5,       6,        7,        8,        9,       10,       11,         12
+		var id, typStr, desc, ownerStr, state, status, created, finished, modified, fraction, resolved, errorMsg, instanceID tree.Datum
+
+		if ok {
+			r := it.Cur()
+			id, typStr, desc, ownerStr, state, status, created, finished, modified, fraction, resolved, errorMsg, instanceID =
+				r[0], r[1], r[2], r[3], r[4], r[5], r[6], r[7], r[8], r[9], r[10], r[11], r[12]
+
+			owner := username.MakeSQLUsernameFromPreNormalizedString(string(tree.MustBeDString(ownerStr)))
+			// TODO(dt): replace this with the more permissive jobsauth.Authorize that
+			// that allows non-owner non-admins access to jobs from other users before
+			// we can make this version on by default.
+			if owner != p.User() && !isAdmin {
+				continue
+			}
+		} else if !ok {
+			if len(sessionJobs) == 0 {
+				return emitted, nil
+			}
+			job := sessionJobs[len(sessionJobs)-1]
+			sessionJobs = sessionJobs[:len(sessionJobs)-1]
+			payloadType, err := jobspb.DetailsType(jobspb.WrapPayloadDetails(job.Details))
+			if err != nil {
+				return emitted, err
+			}
+			// synthesize the fields we'd read from the jobs table if this job were in it.
+			id, typStr, desc, ownerStr, state, status, created, finished, modified, fraction, resolved, errorMsg, instanceID =
+				tree.NewDInt(tree.DInt(job.JobID)),
+				tree.NewDString(payloadType.String()),
+				tree.NewDString(job.Description),
+				tree.NewDString(job.Username.Normalized()),
+				tree.NewDString(string(jobs.StatusRunning)),
```

Comment:
the legacy version marks these session jobs as `StatusPending`.

Reply:
Ah, so _that's_ where we use this status!